com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/HooksPhaseBuilder.java

```diff
@@ -0,0 +1,46 @@
+package com.sap.cloud.lm.sl.cf.process.util;
+
+import com.sap.cloud.lm.sl.cf.core.model.HookPhase;
+import com.sap.cloud.lm.sl.cf.core.model.HookPhaseProcessType;
+import com.sap.cloud.lm.sl.cf.core.model.HookProcessPhase;
+import com.sap.cloud.lm.sl.cf.core.model.Phase;
+import com.sap.cloud.lm.sl.cf.core.model.SubprocessPhase;
+import com.sap.cloud.lm.sl.cf.process.steps.ProcessContext;
+import com.sap.cloud.lm.sl.cf.process.variables.Variables;
+import com.sap.cloud.lm.sl.cf.web.api.model.ProcessType;
+
+public class HooksPhaseBuilder {
+
+    private static final String HOOKS_DELIMITER = ".";
+    private final ProcessTypeParser processTypeParser;
+    private final ProcessContext context;
+
+    public HooksPhaseBuilder(ProcessTypeParser processTypeParser, ProcessContext context) {
+        this.processTypeParser = processTypeParser;
+        this.context = context;
+    }
+
+    public String buildHookPhase(HookPhase hookPhase) {
```

Comment:
I would extract the whole logic as follows - as we are having hook phases which consists of the following parts <deployment-type>.<entity-type>.<phase>.<optional-phase-locator> we could extract this logic in different methods or inner classes and the whole method would look like this:
```
public String buildHookPhase(HookPhase hookPhase){
     String getDeploymentType = getDeploymentType(); // perform some validation whether the current deployment                                                                                                                                                      type is deploy or blue-green
     String optionalPhaseLocator = getOptionalPhaseLocator(); // here the check whether the context.getVariable(Variables.PHASE) != Phase.AFTER_RESUME should be made
     return getDeploymentType + "." + DEFAULT_HOOK_ENTITY + "." + hookPhase.getValue() + "." + optionalPhaseLocator // if presented, else it should be omitted.

}
```